tests/test_femtonics.py

```
@@ -0,0 +1,149 @@
+import numpy as np
+from numpy import dtype
+from numpy.testing import assert_array_equal
+from datetime import datetime, timezone
+
+from roiextractors.extractors.femtonicsimagingextractor import FemtonicsImagingExtractor
+from .setup_paths import OPHYS_DATA_PATH
+
+
+def test_femtonicsimagingextractor_p29_mesc():
+    """
+    Test with Femtonics p29.mesc file.
+
+    File: p29.mesc
+    Metadata (note: metadata vs actual data mismatch):
+    - Source: Femtonics .mesc file (MESc 3.3)
+    - Metadata claims: 56250 samples (frames)
+    - Actual data contains: 5 samples (frames) - this appears to be a preview/subset
+    - Frame rate: ~30.96 Hz (32.297 ms per frame)
+    - Frame shape: (512, 512) pixels
+    - Pixel size: 1.782 µm x 1.782 µm
+    - Channels: 2 (UG, UR)
+    - Experimenter: flaviod
+    - Date: 2017-09-29 09:53:00.903594
+    - Setup: SN20150066006-MBMoser_1
+
+    This test verifies that the extractor correctly:
+    1. Loads the file and extracts metadata (frame count, resolution, channels, etc.).
+    2. Retrieves the correct channels and properties.
+    3. Tests Femtonics-specific functionality like channel selection and metadata extraction.
+    4. Handles metadata vs actual data mismatches gracefully.
+
+    Note : the actual data only contains 5 frames, not the 56250 claimed in metadata likely due to being truncated for testing.
+    """
+    file_path = OPHYS_DATA_PATH / "imaging_datasets" / "Femtonics" / "moser_lab_mec" / "p29.mesc"
+    extractor = FemtonicsImagingExtractor(file_path=file_path)
```

When there is more than one channel, the extractor should throw an error showing the user the options so they can be deliberate about their choice.

---

Adapted the  _setup_channel_selection(self):
        - Gets available channels for the selected MUnit.
        - If no channels, raises an error.
        - If channel_name is not given and multiple channels exist, raises an error listing options.
        - If channel_name is given, checks it exists; otherwise, raises an error.
        - If only one channel, selects it automatically.

logic explanation also added to the function documentation